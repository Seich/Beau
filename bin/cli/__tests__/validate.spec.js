const ValidateCommand = require('../commands/validate');

jest.mock('../utils');

describe('Validate Command', () => {
	let result;

	beforeEach(() => {
		result = [];
		jest
			.spyOn(process.stdout, 'write')
			.mockImplementation(val =>
				result.push(require('strip-ansi')(val.toString('utf8')))
			);
	});

	afterEach(() => jest.restoreAllMocks());

	it('Validate the configuration file', async () => {
		await ValidateCommand.run([]);
		expect(result).toMatchSnapshot();
	});
});
