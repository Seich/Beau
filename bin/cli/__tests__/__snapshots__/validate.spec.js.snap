// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate Command Validate the configuration file 1`] = `
Array [
  "beau.yml is valid.
",
]
`;
